qutip/solver/ode/explicit_rk.pyx

```diff
@@ -0,0 +1,485 @@
+    if type(in_) is Dense:
+        for ptr in range(in_.shape[0] * in_.shape[1]):
+            (<Dense> out).data[ptr] = (<Dense> in_).data[ptr]
+        return out
+    else:
+        return in_.copy()
+
+
+cdef Data iadd_data(Data left, Data right, double complex factor):
+    # left += right * factor
+    # reusing `left' allocated buffer if possible.
+    # TODO: when/if iadd_csr is added: move to data/add.pyx.
+    if factor == 0:
+        return left
+    if type(left) is Dense:
+        iadd_dense(left, right, factor)
+        return left
+    else:
+        return add(left, right, factor)
+
+
+cdef class Explicit_RungeKutta:
+    """
+    Qutip implementation of Runge Kutta ODE.
+    Works in :class:`Data` allowing solving using sparse and gpu data.
+
+    Parameters
+    ----------
+    qevo : QobjEvo
+        The system to integrate: ```dX = qevo.matmul_data(t, X)``
+
+    rtol, atol : double
+        Relative and absolute tolerance of the integration error respectively.
+
+    nsteps : int
+        Maximum number of steps done during one call of integrate.
+
+    first_step : double
+        Lenght in ``t`` of the first step. If ``0``, an appropriate step will be
+        determined automaticaly.
+
+    min_step, max_step : double
+        Bounds of the step lenght in ``t``. ``0`` means no bounds.
+
+    interpolate : bool
+        Whether to do longer step and interpolate back.
+        Example:
+            At ``t=1``, the integrator is asked to integrate to ``t=2`` but can
+            safely integrate up to ``t=3``. If ``interpolate=True``, it will
+            integrate up to ``t=3`` and then use interpolation to get the state
+            at ``t=2``. With ``interpolate=False``, it will integrate to
+            ``t=2``.
+
+    method : ['euler', 'rk4', 'vern7', 'vern9']
+        The integration method to use.
+        * It also accept a tuple of runge-kutta coefficients.
+        (See `Explicit_RungeKutta._init_coeff`)
+    """
+    def __init__(self, QobjEvo qevo, double rtol=1e-6, double atol=1e-8,
+                 int nsteps=1000, double first_step=0, double min_step=0,
+                 double max_step=0, bint interpolate=True, method="euler"):
+        # Function to integrate.
+        self.qevo = qevo
+        # tolerances
+        self.atol = atol
+        self.rtol = rtol
+        self._dt_safe = atol
+        # Number of step to do before giving up
+        self.max_numsteps = nsteps
+
+        # contrain on the step times
+        self.first_step = first_step
+        self.min_step = min_step
+        self.max_step = max_step
+
+        # If True, will do the steps longer that the target and use
+        # interpolation to get the state at the wanted time.
+        self.interpolate = interpolate
+
+        self.k = []
+        if isinstance(method, tuple):
+            self._init_coeff(*method)
+        elif "vern7" == method:
+            self._init_coeff(*vern7_coeff)
+        elif "vern9" == method:
```

Comment:
I wonder if we should name these `vern7-eff` and `vern9-eff` in case we want to implement `vern7-robust` later? Or at least allow `vern7-eff` as an alternative name?